src/ddmd/expressionsem.d

```diff
@@ -331,7 +331,118 @@ extern (C++) Expression resolveProperties(Scope* sc, Expression e)
     return e;
 }
 
+/****************************************
+ * The common type is determined by applying ?: to each pair.
+ * Output:
+ *      exps[]  properties resolved, implicitly cast to common type, rewritten in place
+ *      *pt     if pt is not NULL, set to the common type
+ * Returns:
+ *      true    a semantic error was detected
+ */
+private extern (C++) bool arrayExpressionToCommonType(Scope* sc, Expressions* exps, Type* pt)
```

Comment:
Remove `extern (C++)` - it's clearly not needed as `arrayExpressionToCommonType` can only be called by D code from this module.